### PR TITLE
Feature/marp 188 new connector vertex ai api google

### PR DESCRIPTION
**Pull request by tvtphuc-axonivy**

### CODE REVIEW

**Review by nntthuy-axonivy — APPROVED** [review]
LGTM

File: vertexai-google/config/variables.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# == Variables ==
+# 
+# You can define here your project Variables.
+# If you want to define/override a Variable for a specific Environment, 
+# add an additional ‘variables.yaml’ file in a subdirectory in the ‘Config’ folder: 
+# '<project>/Config/_<environment>/variables.yaml
+#
+
+Variables:
+  gemini:
```
Comment (ivy-rew) [inline vertexai-google/config/variables.yaml:10]: why is there an extra section for 'gemini' while there is already a 'vertex-gemini' part? it seems is if these properties belong together and therefore they should live under the same section.

File: vertexai-google/config/variables.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    apiKey: ''
+  vertexai-gemini:
+    projectId: ''
+    location: ''
+    modelName: ''
+    keyFilePath: ''
```
Comment (ivy-rew) [inline vertexai-google/config/variables.yaml:16]: these properties come with zero description. but there is some information in the readme.md how to use them.

note that variables should never be without accurate description in market products. these variables and their description are prominently shown throughout the product (variables-editor in designer and engine-cockpit) ... while the readme is soon forgotten after the installation. so the most important piece of information should also be stated here.

File: vertexai-google/config/variables.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# If you want to define/override a Variable for a specific Environment, 
+# add an additional ‘variables.yaml’ file in a subdirectory in the ‘Config’ folder: 
+# '<project>/Config/_<environment>/variables.yaml
+#
+
+Variables:
+  gemini:
+    apiKey: ''
+  vertexai-gemini:
+    projectId: ''
```
Comment (ivy-rew) [inline vertexai-google/config/variables.yaml:13]: state an example project id. is it a UUID?

File: vertexai-google/config/variables.yaml (hunk 2 above)
Comment (ivy-rew) [inline vertexai-google/config/variables.yaml:16]: keyFilePath: absolute? relative? both possible? where should I place my keyFiles? please guide the user to use 'configuration' dir of the engine if feasible.

**Review by ivy-rew — COMMENTED** [review]
@tvtphuc-axonivy please update the docs for this connector, or it will be hard to setup and prone to errors.

**Comment by ivy-rew** [comment]
> LGTM

we maintain a check-list for new market products that we verify during reviews. 
it seems to me that the point about accurate variables description was completely forgotten. 
please do careful reviews

File: vertexai-google/config/variables.yaml (hunk 2 above)
Comment (tvtphuc-axonivy) [reply vertexai-google/config/variables.yaml:16]: The keyFilePath apply for both absolute and relative url. We just need to set correct file path in the variable

File: vertexai-google/config/variables.yaml (hunk 3 above)
Comment (tvtphuc-axonivy) [reply vertexai-google/config/variables.yaml:13]: No , this projectId get from the project that user create in the google cloud

File: vertexai-google/config/variables.yaml (hunk 2 above)
Comment (tvtphuc-axonivy) [reply vertexai-google/config/variables.yaml:16]: I have added some comments in Variables , also the example of each variable.
Can you help me to review this PR ?
https://github.com/axonivy-market/vertexai-google/pull/14/files

File: vertexai-google/config/variables.yaml (hunk 1 above)
Comment (tvtphuc-axonivy) [reply vertexai-google/config/variables.yaml:10]: In this connector, there are two ways to use Gemini (we can switch options together):
1. The variable "gemini" is used to call the Gemini model directly.
2. The variable "vertex-gemini" is used to call the Gemini model through the Vertex Platform.
The response between Gemini and vertex-gemini is not too different.

Comment (ivy-rew) [reply vertexai-google/config/variables.yaml:10]: ok for me then; keep it as you prefer.

**Pull request merged**